timely_beliefs/beliefs/utils.py

```diff
@@ -568,10 +589,16 @@ def read_csv(
         raise TypeError(
             f"Extension {ext} not recognized. Accepted file extensions are csv, xlsx and xls."
         )
+    if filter_by_column:
+        # Filter the read-in data
+        for col, val in filter_by_column.items():
+            df = df[df[col] == val]
+        # Remove the columns used to filter
+        df = df.drop(columns=filter_by_column.keys())
```

Comment:
This should only happen if they are not in `use_cols`.

Reply:
Nice catch. That would allow to filter not just by an extra column, but also by a column that is part of the beliefs data.